Sources/NetShears/Storage.swift

```diff
@@ -9,40 +9,49 @@
 import Foundation
 
 final class Storage: NSObject {
-
     static let shared: Storage = Storage()
-    
+    private let accessQueue = DispatchQueue(label: "com.netshears.queue", attributes: .concurrent)
+
     private(set) var requests: [NetShearsRequestModel] = []
 
     var filteredRequests: [NetShearsRequestModel] {
-        return getFilteredRequests()
+        getFilteredRequests()
     }
 
-    func saveRequest(request: NetShearsRequestModel?){
-        guard request != nil else {
-            return
-        }
-        
-        if let index = requests.firstIndex(where: { (req) -> Bool in
-            return request?.id == req.id ? true : false
-        }) {
-            requests[index] = request!
-        } else {
-            requests.insert(request!, at: 0)
+    func saveRequest(request: NetShearsRequestModel) {
+        accessQueue.async(flags: .barrier) { [weak self] in
+            guard let self else {
```

Comment:
Doesn't it limit NetShears to get compiled on Swift 5.7 or later? I guess we can have it with legacy syntax.
```swift
guard let self = self else { return }
```